test/functorch/test_ac.py

```diff
@@ -374,6 +375,33 @@ def call():
         try_seq_length(4, 7, "mm")
         try_seq_length(4, 9, "attn")
 
+    def test_manual_ac(self):
+        # test that manual checkpoint boundaries are respected
+        # when autoac is set
+        def f(x):
+            tmp1 = torch.matmul(x, x.T)
+            tmp1 = torch.matmul(tmp1, tmp1)
+            tmp1 = torch.matmul(tmp1, tmp1)
+            out = torch.matmul(tmp1, x)
+            return out
+
+        def g(x):
+            x = checkpoint(f, x, use_reentrant=False)
+            x = checkpoint(f, x, use_reentrant=False)
+            return x
+
+        x = torch.randn(64, 1024, requires_grad=True)
+
+        def call():
+            return g(x).sum()
+
+        eager_mem, eager_flops = get_mem_and_flops(call)
```

Comment:
oh really nice idea using flops as our test for whether we are respecting recompute decisions!